src/AddControlDialog.h

```diff
@@ -0,0 +1,37 @@
+#ifndef CONTROLDIALOG_OPENMSX_H
+#define CONTROLDIALOG_OPENMSX_H
+
+#include "ui_ControlDialog.h"
+#include <QDialog>
+
+class ControlDialog : public QDialog, private Ui::ControlDialog
+{
+	Q_OBJECT
+public:
+	ControlDialog(QWidget* parent = nullptr);
+
+    QString icon() const;
+    QString name() const;
+    QString command() const;
+    QString function() const;
```

Comment:
I know, still in development ...

I guess these should return `const QString&`?
I know QString is internally reference counted, so copies are cheap. But no copy is even cheaper. And it's a good habit anyway.